docs/architecture/grpc-for-wcf-developers/why-grpc.md

```diff
@@ -1,11 +1,45 @@
 ---
 title: Why gRPC is recommended for WCF developers - gRPC for WCF Developers
-description: TO BE WRITTEN
+description: A discussion of why gRPC is a good fit for WCF developers looking to migrate to modern architectures and platforms.
 author: markrendle
 ms.date: 09/02/2019
 ---
 
 # Why gRPC is recommended for WCF developers
 
+Before diving deeply into the language and techniques of gRPC, it's worth discussing why gRPC is the right solution for WCF developers who want to migrate to .NET Core, given there are alternatives available.
+
+## Similarity to WCF
+
+Although its implementation and approach is different, the actual experience of developing and consuming services with gRPC should be very intuitive for WCF developers. The underlying goal of making it possible to code as though the client and server were on the same platform, without needing to worry about networking, is the same. Both platforms share the principle of declaring and then implementing an interface, even though the process for declaring that interface is different. And as you'll see in chapter 5, the different types of RPC calls supported by gRPC map very well to the different bindings available to WCF services.
+
+## Benefits of gRPC
+
+Additional reasons why gRPC stands above other solutions are:
+
+### Performance
+
+As already discussed, using HTTP/2 rather than HTTP/1.1 removes the requirement for human-readable messages and instead uses the smaller faster binary protocol. This is more efficient for computers to parse. HTTP/2 also supports multiplexing requests over a single connection enabling responses to be sent as soon as they're ready without the need to wait in a queue (an issue in HTTP/1.1 known as "head-of-line (HOL) blocking"). Fewer resources are needed when using gRPC, which makes it a good solution to use for mobile devices and over slower networks.
+
+### Interoperability
+
+There are gRPC tools and libraries for all major programming languages and platforms, including .NET, Java, Python, Go, C++, Node.js, Swift, Dart, Ruby, and PHP. Thanks to the Protocol Buffers binary wire format and the efficient code generation for each platform, developers can achieve better performance than WCF's NetTCP bindings while still enjoying full cross-platform support.
+
+### Usability and productivity
+
+gRPC is a comprehensive RPC solution. It has thoroughly addressed the full range of networking and connectivity concerns. gRPC works consistently across multiple languages. gRPC provides excellent tooling, with much of the boilerplate auto-generated, so more developer time is freed up to focus on business logic.
```

Comment:
I think the phrase "thoroughly addressed the full range of networking and connectivity concerns" is going to result in some negative pushback because of how vague it is. What does "the full range of networking and connectivity concerns" include? Reliable session solves the problem of your connection dropping and retrying without the server processing the request twice. I would say this is included in "connectivity concerns" but gRPC doesn't solve that problem. It also doesn't solve the problem of ensuring in-order execution of requests. You should be able to send multiple request serially without waiting for the prior response to complete (think high latency network) but guarantee they execute in order serially on the service. gRPC can't do this out of the box.